src/Perlang.Interpreter/Typing/TypeCoercer.cs

```diff
@@ -76,7 +80,7 @@ public static bool CanBeCoercedInto(Type targetType, Type sourceType)
             // TODO: Implement more of the coercions being advertised in the XML docs. :)
 
             int? sourceSize = SignedIntegerLengthByType.TryGetObjectValue(sourceType);
-            int? targetSize = SignedIntegerLengthByType.TryGetObjectValue(targetType);
+            int? targetSize = SignedIntegerLengthByType.TryGetObjectValue(targetType) ?? FloatIntegerLengthByType.TryGetObjectValue(targetType);
```

Comment:
Then, we can use it here. Note that it is **specifically** only used for the target type; conversion from `int` to `double` is valid but not the other way around.